### PR TITLE
Logics Modified.! Othello Running fine. 

### PR DESCRIPTION
**Pull request by sujithTSR**
TODO: Winning cases when all empty spaces are not filled, are to be added.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -191,9 +200,15 @@ function possibleMoves(squares, player, i, j) {
 function getPossibleMovesInDirection(squares, direction, i, j, player) {
   const oppPlayer = player === 'X' ? 'O' : 'X';
   const directionMoves = [];
+  if (getDirectionIndex(direction, i, j) == null ){
+    return [];
+  }
   [i, j] = getDirectionIndex(direction, i, j);
   while (squares[i][j] === oppPlayer && i !== null && j !== null) {
     directionMoves.push([i,j]);
+    if (getDirectionIndex(direction, i, j) === null){
```
Comment (amishas157) [inline src/index.js:209]: Think that this condition is being handled by https://github.com/amishas157/reversi-react-electron/pull/1/files#diff-1fdf421c05c1140f6d71444ea2b27638R214

```
 if (i && j && squares[i][j] === player) {
```

If either of `i` or `j` is `null`, `[]` will be returned

File: src/index.js [hunk 2]
```diff
@@ -32,26 +32,30 @@ class Board extends React.Component {
...
     const player = this.state.xIsNext ? 'X' : 'O';
     const moves = possibleMoves(squares, player, i, j)
-
     console.log(moves);
-
-    const reverseElements = moves.reduce((move) => {
-      return move.length !== 0 ? move : false;
+    console.log("sujith");
+    
+    const reverseElements = [];
```
Comment (amishas157) [inline src/index.js:38]: Whoops, didn't really understand this part. Can you explain a bit?

File: src/index.js [hunk 3]
```diff
@@ -202,42 +217,71 @@ function getPossibleMovesInDirection(squares, direction, i, j, player) {
...
-    const [a, b, c] = lines[i];
-    if (squares[a] && squares[a] === squares[b] && squares[a] === squares[c]) {
-      return squares[a];
+function get_player_counts(squares){
+  let x_count = 0;
+  let o_count = 0;
+
+  for (let i_iter = 0; i_iter < 8 ; i_iter++ ){
+    for (let j_iter = 0;j_iter<8; j_iter++){
+      if (squares[i_iter][j_iter] !== null && squares[i_iter][j_iter] !== 'X' ){
```
Comment (amishas157) [inline src/index.js]: Might need to change this condition 
from `squares[i_iter][j_iter] !== 'X'`
to `squares[i_iter][j_iter] === 'X'`

That's why it is giving reverse scores 😛. Same for next condition

File: src/index.js [hunk 4]
```diff
@@ -202,42 +217,71 @@ function getPossibleMovesInDirection(squares, direction, i, j, player) {
...
+      else if (squares[i_iter][j_iter] !== null && squares[i_iter][j_iter] !== 'O' ){
+        o_count++;
+      }
+    }
+  }
+  return [x_count,o_count];
+}
+
+function calculateWinner(x_count, o_count, squares) {
+  // Winner can be calculated with the one having more number of places
```
Comment (amishas157) [inline src/index.js:259]: Yes, right. This was a winner logic from tic-tac-toe and therefore should be changed. Copy pasta. :D

**Review by amishas157 — CHANGES_REQUESTED** [review]
Looks great to me. Thanks for it 🙇‍♀️ . Also, left some comments and questions, once addressed, let's merge this.

**Comment by amishas157** [comment]
Yay 🤗

**Comment by sujithTSR** [comment]
Great.. (y)

**Pull request merged**